### ISSUE TITLE
Why is pageYOffset marked as deprecated?

### ISSUE THREAD

**Issue by chandraaditya** [microsoft/TypeScript-DOM-lib-generator#1642]
`scrollY` is an alias of `pageYOffset`, but there's no documentation anywhere else (as far as I'm aware) that `pageYOffset` is deprecated or that it is a legacy api, but it is marked as deprecated in the definitions.

Why was [this](https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1082#pullrequestreview-719665841) decision made? 

References to `pageYOffset` and `scrollY`:
https://developer.mozilla.org/en-US/docs/web/api/window/pageyoffset
https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollY

These references also say that `pageYOffset` has more support than `scrollY` in older browsers, and I don't seem to find any discussion about deprecating it or saying that `pageYOffset` is a legacy API os `scrollY`, all the documentations say they are both equal.

The main issue I am having with this is that IntelliJ uses these definitions in the IDE and so it's being marked as deprecated [link to issue](https://youtrack.jetbrains.com/issue/WEB-55890/wrong-deprecated-warning-for-window.pageXoffset-and-window.pageYoffset).

(I created an issue [here](https://github.com/microsoft/TypeScript/issues/56308) and they recommended I move it here.)

**Comment by saschanaz** [comment]
Hmm.

1. If you closely look at #1082, you can see that it's not about a decision for specific properties but about reading MDN data and add deprecated tags based on it.
2. The data is using MDN's `alternative_name` (which is mapped as `preferred_name` there), you can see from the compatibility table of scrollX's MDN page has pageXOffset for Chrome for example.
3. That field is almost exclusively used for deprecated prefixed APIs, but later MDN got separate pages for pageX/YOffset with separate compat tables. And then they forgot to remove it from the table of scrollX/Y.
4. So TS still has them marked as deprecated.

tl;dr: MDN should remove alternative_name from the tables of scrollX/Y.

**Comment by chandraaditya** [comment]
@saschanaz but isn't it true that pageX/YOffset is an alternative name for scrollX/Y, at the same time it's not deprecated? Shouldn't they be marked deprecated ONLY and ONLY if they have been marked as deprecated if MDN also marks it as deprecated?

**Comment by saschanaz** [comment]
There's no need for alternative_name when there's a separate table, it's fully redundant.

The thing is, MDN generally only has data for the original API instead of the aliases (see https://developer.mozilla.org/en-US/docs/Web/API/DOMMatrix for example), so there's no "if and only if". Truly deprecated aliases just don't have the data at all.

**Comment by teamolhuang** [comment]
Please just delete this thing because our customer opens up their IDE and asks why we using deprecated properties. This thing shows up in Microsoft Learn too. I posted MDN and they objects with that. If MDN didn't imply it being deprecated, then it means don't fucking use that field to generate deprecated annotations  right into the API because that's wrong data and it will falsely shows up in so many places like Microsoft's document page.

I really want to kill myself because we already have lots on our hands and the issues keep growing. Please just have mercy and fix this shit from two years ago.

**Comment by saschanaz** [comment]
MDN decided to remove pages for pageOffsetX/Y in favor of scrollX/Y. I'd like to follow the decision unless there's a strong reason not to.

@github-actions close

**Comment by monolithed** [comment]
@saschanaz, you are the only person on the internet who decided to mark the pageYOffset property as legacy and deprecated. Even MDN, which you reference, states that pageYOffset is merely an alias for scrollY and nothing more. Furthermore, MDN is not an authoritative source to reference — it's simply a blog. Please provide a source that states pageYOffset is legacy or deprecated, otherwise, roll back your commit.

Here's the link to the specification: https://www.w3.org/TR/cssom-view-1/#dom-window-pageyoffset
And here's the link to the latest draft: https://drafts.csswg.org/cssom-view/#dom-window-pageyoffset

If you wish for this property to become deprecated, join the working group of specification authors and convince them of your point of view. But for now, this property has the broadest support among browsers and, by all appearances, will outlast the dinosaurs.

**Comment by HolgerJeromin** [comment]
Please keep the tone friendly.

**Comment by saschanaz** [comment]
Yes, please.

Maybe this should be special cased, all MDN aliases are currently considered as deprecated ones, and frequently it's clear from the names e.g. WebKitCSSMatrix. This one is arguably in a different situation.
